src/main/java/org/apache/commons/codec/binary/BaseNCodec.java

```diff
@@ -711,12 +711,15 @@ public boolean isStrictDecoding() {
      * @return The number of bytes successfully extracted into the provided byte[] array.
      */
     int readResults(final byte[] b, final int bPos, final int bAvail, final Context context) {
-        if (context.buffer != null) {
+        if (hasData(context)) {
             final int len = Math.min(available(context), bAvail);
             System.arraycopy(context.buffer, context.readPos, b, bPos, len);
             context.readPos += len;
             if (context.readPos >= context.pos) {
```

Comment:
Perhaps this would be clearer as:
```java
if (!hasData(context))
```
Reusing the hasData method makes it the single point of entry to the logic with the pos and readPos markers.
The comment can then be changed to:
```java
// All data read.
// Reset position markers but do not set buffer to null to allow its reuse.
// hasData(context) will still return false, and this method will return 0 until
// more data is available, or -1 if EOF.
```